MSMAll/scripts/MSMAll.sh

```diff
@@ -1,1195 +1,254 @@
-				p_RSNTargetFileOrig=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--rsn-cost-weights=*)
-				p_RSNCostWeightsOrig=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--myelin-target-file=*)
-				p_MyelinTargetFile=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--topography-roi-file=*)
-				p_TopographyROIFile=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--topography-target-file=*)
-				p_TopographyTargetFile=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--iterations=*)
-				p_Iterations=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--method=*)
-				p_Method=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--use-migp=*)
-				p_UseMIGP=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--ica-dim=*)
-				p_ICAdim=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--regression-params=*)
-				p_RegressionParams=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--vn=*)
-				p_VN=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--rerun=*)
-				p_ReRun=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--reg-conf=*)
-				p_RegConf=${argument/*=/""}
-				index=$(( index + 1 ))
-				;;
-			--reg-conf-vars=*)
-				# Note: since the value of this parameter contains equal signs ("="),
-				# we have to handle grabbing the value slightly differently than
-				# in the other cases.
-				p_RegConfVars=${argument#--reg-conf-vars=}
-				index=$(( index + 1 ))
-				;;
-			--matlab-run-mode=*)
-				p_MatlabRunMode=${argument#*=}
-				index=$(( index + 1 ))
-				;;
-			*)
-				show_usage
-				log_Err_Abort "unrecognized option: ${argument}"
-				;;
-		esac
-	done
+Iterations=$(echo "${Iterations}" | sed 's/_/ /g')
+NumIterations=$(echo "${Iterations}" | wc -w)
+CorrectionSigma=$(echo "sqrt ( 200 )" | bc -l)
+BC="NO"
+nTPsForSpectra="0" #Set to zero to not compute spectra
+VolParams="NO" #Dont' output volume RSN maps
+
+# Log values of Naming Conventions and other variables
+log_Msg "Caret7_Command: ${Caret7_Command}"
+log_Msg "AtlasFolder: ${AtlasFolder}"
+log_Msg "DownSampleFolder: ${DownSampleFolder}"
+log_Msg "NativeFolder: ${NativeFolder}"
+log_Msg "ResultsFolder: ${ResultsFolder}"
+log_Msg "T1wFolder: ${T1wFolder}"
+log_Msg "DownSampleT1wFolder: ${DownSampleT1wFolder}"
+log_Msg "NativeT1wFolder: ${NativeT1wFolder}"
+log_Msg "LowICAdims: ${LowICAdims}"
+log_Msg "Iterations: ${Iterations}"
+log_Msg "NumIterations: ${NumIterations}"
+log_Msg "CorrectionSigma: ${CorrectionSigma}"
+log_Msg "BC: ${BC}"
+log_Msg "nTPsForSpectra: ${nTPsForSpectra}"
+log_Msg "VolParams: ${VolParams}"
+
+IndArealDistortionFile=${NativeFolder}/${Subject}.ArealDistortion_${RegNameStem}_${NumIterations}_d${ICAdim}_${Method}.native.dscalar.nii
+if [[ -e ${IndArealDistortionFile} &&  ${ReRun} = "NO" ]]; then
+	log_Msg "IndArealDistortionFile exists: ${IndArealDistortionFile} and ReRun is disabled"
+	log_Msg "the main functionality of MSMAll.sh script is finished"
+	exit 0
```

Comment:
This is a much better solution for this behavior than the previous `if` statement over the entire function body.  However, the log message should perhaps say "--rerun is set to 'no'" and "skipping MSMAll.sh" rather than saying the functionality is finished.

Also, please use `opts_StringToBool` on `ReRun`, to also accept true/false, 0/1, and all capitalizations.  The previous implementation was "literal `YES`, capitalized, everything else means no, no sanity checking", which is a little unfortunate, but the current PR instead treats anything ambiguous as a yes, which could bite unsuspecting users.